action.yml

```diff
@@ -108,12 +108,51 @@ runs:
         ref: ''
         path: ".jit/"
 
-    - name: Checkout original repository
+    - name: Sparse checkout repository
+      continue-on-error: true
+      id: sparse-checkout-repository
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ steps.oidc-auth.outputs.github_token }}
+      # only checkout if control's workflow job defines it and if there is a base and head sha to compare to
+      if: >-
+        ( !(fromJSON(github.event.inputs.client_payload).payload.runner_setup) || fromJSON(github.event.inputs.client_payload).payload.runner_setup.checkout == true )
```

Comment:
It's good to include here the branch in the IF as well, it's more explicit that we must have it